### PR TITLE
DateBox: Rearrange attaching mouseEvents before rendering masks

### PR DESCRIPTION
**Pull request by nikkithelegendarypokemonster**

### CODE REVIEW

File: packages/devextreme/js/__internal/ui/date_box/m_date_box.mask.parts.ts [hunk 1]
```diff
@@ -3,6 +3,17 @@ import { extend } from '@js/core/utils/extend';
 import { fitIntoRange } from '@js/core/utils/math';
 import { getPatternSetters } from '@js/localization/ldml/date.parser';
 
+interface Section {
```
Comment (EugeniyKiyashko) [inline packages/devextreme/js/__internal/ui/date_box/m_date_box.mask.parts.ts]: Can we revert this changes from the PR?

File: packages/devextreme/js/__internal/ui/date_box/m_time_view.ts [hunk 2]
```diff
@@ -9,6 +9,13 @@ import SelectBox from '@js/ui/select_box';
 
 import dateUtils from './m_date_utils';
 
+interface Item {
```
Comment (EugeniyKiyashko) [inline packages/devextreme/js/__internal/ui/date_box/m_time_view.ts]: We should use other way to define type of Item. We can discuss it, but let's revert all changes in PR that doesn't relate with bug fix. It allow us to cherry pick changes in previous versions without additional difficulties.

File: packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/datebox.mask.tests.js [hunk 3]
```diff
@@ -1004,6 +1004,33 @@ module('Events', setupModule, () => {
 
         assert.strictEqual(event.isDefaultPrevented(), true, `the ${DROP_EVENT_NAME} event is prevented`);
     });
+
+    QUnit.test('change year via scroll should log proper year in on value change event (T1229926)', function(assert) {
```
Comment (EugeniyKiyashko) [inline packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/datebox.mask.tests.js]: Your test is always green. Before calling for a review, ensure that the test reproduces the scenario.

Comment (EugeniyKiyashko) [inline packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/datebox.mask.tests.js]: We can use this variant:

```
QUnit.test('change year via scroll should log proper year in on value change event (T1229926)', function(assert) {
        const valueChangedHandler = sinon.spy();
        const date = new Date();
        const currentYear = date.getFullYear();

        const datebox = $('#dateBox').dxDateBox({
            type: 'date',
            value: date,
            displayFormat: 'M/dd/yyyy',
            valueChangeEvent: 'dxmousewheel',
            useMaskBehavior: true,
            onValueChanged: valueChangedHandler
        }).dxDateBox('instance');
        
        const $input = datebox.$element().find(`.${TEXTEDITOR_INPUT_CLASS}`);

        keyboardMock($input, true)
            .caret({ start: 12, end: 15 });

        $($input).trigger('dxclick');

        pointerMock($input)
            .wheel(1);

        const value = datebox.option('value');

        let changedValue = valueChangedHandler.getCall(0).args[0];
        
        assert.strictEqual(valueChangedHandler.callCount, 1, 'handler has been called once');
        assert.deepEqual(new Date(changedValue.value).getFullYear(), currentYear + 1, 'value year is correct');
        assert.deepEqual(new Date(changedValue.previousValue).getFullYear(), currentYear, 'previous value year is correct');
        
        pointerMock($input)
            .wheel(1);

        changedValue = valueChangedHandler.getCall(1).args[0];

        assert.strictEqual(valueChangedHandler.callCount, 2, 'handler has been called twice');
        assert.deepEqual(new Date(changedValue.value).getFullYear(), currentYear + 2, 'value year is correct');
        assert.deepEqual(new Date(changedValue.previousValue).getFullYear(), currentYear + 1, 'previous value year is correct');
    });
```

We should place this test in `datebox.tests.js`.

Comment (nikkithelegendarypokemonster) [reply packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/datebox.mask.tests.js]: [889bce5](https://github.com/DevExpress/DevExtreme/pull/27418/commits/889bce591bb90b93070b93c7ab94ad1047175923)

Comment (nikkithelegendarypokemonster) [reply packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/datebox.mask.tests.js]: [889bce5](https://github.com/DevExpress/DevExtreme/pull/27418/commits/889bce591bb90b93070b93c7ab94ad1047175923)

**Review by EugeniyKiyashko — APPROVED** [review]

**Review by Zedwag — APPROVED** [review]
